Sources/App/Settings/SettingsDetailViewController.swift

```diff
@@ -444,6 +445,28 @@ class SettingsDetailViewController: HAFormViewController, TypedRowControllerType
 
             let scenes = realm.objects(RLMScene.self).sorted(byKeyPath: RLMScene.positionKeyPath)
 
+            let toggleAllSwitch = SwitchRow()
+            toggleAllSwitch.title = L10n.SettingsDetails.Actions.Scenes.selectAll
+
+            scenesUpdateObserver = scenes.observe { _ in
+                // Access UISwitch directly to set state to avoid triggering "on value change"
```

Comment:
what happens if you change the model instead, and that triggers a cell update?

Reply:
Any changes on scenes model should result in this block of code being executed and updating the "select all" toggle state, is this what you mean or did I misunderstand?

Reply:
Don't put this in one of the model update blocks. Create the model, set up this observer to edit the model, and set the callbacks on the model for when the user toggles it.

Reply:
I don't think "Select all" should be part of the model, it's just a visual representation that reacts according to the group of scenes state.

Reply:
I updated the PR based on our talk in discord, also I moved the cell a little bit:
![Simulator Screenshot - iPhone 15 Pro - 2023-11-07 at 16 49 07](https://github.com/home-assistant/iOS/assets/5808343/04601fdc-8243-4ee4-942b-78203614663e)